### ISSUE TITLE
Consider implementing an OID-based decoder compatibility check

### ISSUE THREAD

**Issue by nikita-volkov** [nikita-volkov/hasql#97]

**Comment by ocharles** [comment]
`postgresql-simple` does this, which you might want to use for reference.

**Comment by nikita-volkov** [comment]
The only usefulness of that would be in catching only a portion of possible errors and doing so in runtime. To me the benefit of that seems arguable.

**Comment by ocharles** [comment]
Personally I take the philosophy that most bugs lie at the boundaries of systems, and `hasql` is precisely there to interface between two systems. This is where a lot of error checking should be done, so I'd be in favour of doing this check.

**Comment by nikita-volkov** [comment]
I agree. However implementation-wise, it's gonna take some effort. I'll schedule doing this in the libpq-free version

**Comment by bergey** [comment]
Do you think checking the type info is easier if you bypass libpq?  Or is your schedule just about prioritization?

**Comment by nikita-volkov** [comment]
> Do you think checking the type info is easier if you bypass libpq?

No. It's perfectly possible to get this data thru libpq. And I would accept a PR implementing this.

> Or is your schedule just about prioritization?

It is.
